service/controller/chart/resource/release/desired.go

```diff
@@ -31,18 +33,20 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interf
 	}
 
 	// Merge configmap and secret to provide a single set of values to Helm.
-	values, err := helmclient.MergeValues(configMapData, secretData)
+	err = mergo.Merge(&configMapData, secretData, mergo.WithOverride)
 	if err != nil {
 		return nil, microerror.Mask(err)
 	}
 
+	convertFloat(configMapData)
```

Comment:
I think this needs a comment to explain why we need to convert the data.

Also can we be sure that converting the floats to integers won't cause problems in the components using these values?

Reply:
Yes, I checked merged values from control-plane apps, and floating values such as `memory_size_gb` are the same as before. 
```
 325               i3.xlarge:
 326                 cpu_cores: 4
 327                 description: I3 Storage Optimized Extra Large
 328                 memory_size_gb: 30.5
 329                 storage_size_gb: 950
 330               m3.2xlarge:
 331                 cpu_cores: 8
 332                 description: M3 General Purpose Double Extra Large
 333                 memory_size_gb: 30
 334                 storage_size_gb: 80
 335               m3.large:
 336                 cpu_cores: 2
 337                 description: M3 General Purpose Large
 338                 memory_size_gb: 7.5
 339                 storage_size_gb: 32
```
Other than that, integer values are not changed at all.

Reply:
Ah OK  I think we get the EC2 instance data from an external API somehow.

Thanks, good news its just that!